src/renderers/dom/client/wrappers/__tests__/ReactDOMTextarea-test.js

```diff
@@ -178,27 +223,54 @@ describe('ReactDOMTextarea', function() {
 
     var container = document.createElement('div');
     var stub = <textarea>giraffe</textarea>;
-    stub = renderTextarea(stub, container);
-    var node = ReactDOM.findDOMNode(stub);
+    var node = renderTextarea(stub, container, true);
 
     expect(console.error.argsForCall.length).toBe(1);
     expect(node.value).toBe('giraffe');
 
-    // Changing children should do nothing, it functions like `defaultValue`.
+    // Changing children should cause value to change (new behavior of `defaultValue`)
     stub = ReactDOM.render(<textarea>gorilla</textarea>, container);
-    expect(node.value).toEqual('giraffe');
+    expect(node.value).toEqual('gorilla');
```

Comment:
Hmm this is a behavior change i hadn't anticipated. I was thinking that this PR would be unobservable if you don't have a reset button (or CSS selectors based on the value attribute, etc) but not if this is the behavior. We'll have to think on this probably. Even if this is desirable overall it could break people who expect the other behavior so we need to be cautious.

cc @zpao – what do you think? In order to make `<input type="reset" />` work reasonably, this PR changes the value attribute whenever `defaultValue` is set (equivalent to the `.defaultValue` property), but if there hasn't been any user interaction then doing so also changes the *actual* value.